lib/IDE/Refactoring.cpp

```diff
@@ -2602,6 +2602,154 @@ bool RefactoringActionLocalizeString::performChange() {
+    return;
+  
+  EditConsumer.accept(SM, startLocation, "///\n///  - Returns: <#");
+  EditConsumer.accept(SM, startLocation, returnType.getString());
+  EditConsumer.accept(SM, startLocation, "#>\n");
+}
+    
+void processThrows(swift::SourceLoc startLocation,
+                   SourceEditConsumer &EditConsumer,
+                   SourceManager &SM) {
+    
+  EditConsumer.accept(SM, startLocation, "///\n///  - Throws: <#Throws#>\n");
+}
+
+void generateDocCommentForFunction(FuncDecl *funcDecl,
+                                   swift::SourceLoc startLocation,
+                                   SourceEditConsumer &EditConsumer,
+                                   SourceManager &SM) {
+  
+  EditConsumer.accept(SM, startLocation, "/// <#Function Summary#>\n");
+  
+  processParameters(funcDecl->getParameterLists(), startLocation,
+                    EditConsumer, SM);
+  
+  if (funcDecl->hasThrows()) {
+    processThrows(startLocation, EditConsumer, SM);
+  }
+  
+  Type returnType = funcDecl->getReturnTypeLoc().getType();
+  processReturn(returnType, startLocation, EditConsumer, SM);
+  
+  EditConsumer.accept(SM, startLocation,"///\n");
+}
+    
+void generateDocCommentForConstructor(ConstructorDecl *constructorDecl,
+                                      swift::SourceLoc startLocation,
+                                      SourceEditConsumer &EditConsumer,
+                                      SourceManager &SM) {
+  
+  EditConsumer.accept(SM, startLocation, "/// <#Initializer Summary#>\n");
+  
+  processParameters(constructorDecl->getParameterLists(), startLocation, EditConsumer, SM);
+  
+  if (constructorDecl->hasThrows()) {
+    processThrows(startLocation, EditConsumer, SM);
+  }
+  
+  EditConsumer.accept(SM, startLocation,"///\n");
+}
+    
+void generateDocCommentForSubscript(SubscriptDecl *subscriptDecl,
+                                    swift::SourceLoc startLocation,
+                                    SourceEditConsumer &EditConsumer,
+                                    SourceManager &SM) {
+  
+  EditConsumer.accept(SM, startLocation, "/// <#Subscript Summary#>\n");
+  
+  processParameters(subscriptDecl->getIndices(), startLocation,
+                    EditConsumer, SM);
+  
+  Type returnType = subscriptDecl->getElementTypeLoc().getType();
+  processReturn(returnType, startLocation, EditConsumer, SM);
+  
+  EditConsumer.accept(SM, startLocation,"///\n");
+}
+   
+bool canGenerateDocComment(DeclKind kind) {
+  return kind == DeclKind::Func || kind == DeclKind::Constructor ||
+    kind == DeclKind::Subscript;
+}
+    
+bool RefactoringActionDocCommentBoilerplate::
+  isApplicable(ResolvedCursorInfo Tok, DiagnosticEngine &Diag) {
+  if (Tok.Kind != CursorInfoKind::ValueRef)
+    return false;
+  
+  DeclKind tokenKind = Tok.ValueD->getKind();
+  return canGenerateDocComment(tokenKind);
+}
+    
+bool RefactoringActionDocCommentBoilerplate::performChange() {
+  
+  if (!CursorInfo.ValueD)
```

Comment:
For the same reason, change this condition to `if (!CursorInfo.ValueD || CursorInfo.IsRef)`

Reply:
Correct me if I am wrong, is ValueRef a combination of Value and Ref type and that is why we are checking against the IsRef property?